firstvoices/backend/admin/base_admin.py

```diff
@@ -170,3 +172,35 @@ class HiddenBaseAdmin(BaseAdmin):
 
     def has_module_permission(self, request):
         return {}
+
+
+class FilterAutocompleteBySiteMixin(admin.ModelAdmin):
+    def get_search_results(
+        self, request, queryset, search_term, referer_models_list=None
+    ):
+        queryset, use_distinct = super().get_search_results(
+            request, queryset, search_term
+        )
+        if referer_models_list is None or request.path != "/admin/autocomplete/":
+            return queryset, use_distinct
+        # Filter the queryset by site (used for autocomplete_fields in change and add forms)
+        referer_list = request.headers.get("Referer").split("/")
```

Comment:
Why use the referer here? Seems like this will break if anyone follows a direct link, or clicks via a strange route

Reply:
We chatted in real life and this seems fine for now